src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java

```diff
@@ -0,0 +1,116 @@
+      informer.start();
+   }
+
+   public void stop(@Observes ShutdownEvent ignore) {
+      if (informer != null) {
+         log.info("Shutdown informer");
+         informer.close();
+      }
+   }
+
+   private static final class ConfigMapEventHandler implements ResourceEventHandler<ConfigMap> {
+      private interface SendEventFunc{
+         public void sendEvent(String name, EagerCacheRuleSpec rule);
+      }
+      private void processConfigMapAndSend(Set<Map.Entry<String,String>> entries, SendEventFunc f) {
+         for (Entry<String, String> entry : entries) {
+            Builder eRuleBuilder = EagerCacheRuleSpec.newBuilder();
+            try {
+               JsonFormat.parser().ignoringUnknownFields().merge(entry.getValue(), eRuleBuilder);
+            } catch (InvalidProtocolBufferException e) {
+               log.error("Cannot parse eager rule with name {}", entry.getKey());
+               log.debug("{}",e);
+            }
+            f.sendEvent(entry.getKey(), eRuleBuilder.build());
+         }
+      }
+
+      @Override
+      public void onAdd(ConfigMap obj) {
+         processConfigMapAndSend(obj.getData().entrySet(), (name,rule) -> {log.info("calling DynamicRuleManagement.addRule({},{}", name, rule.toString());});         
+      }
+
+      @Override
+      public void onUpdate(ConfigMap oldObj, ConfigMap newObj) {
+         Map<String, String> oldMap =  oldObj.getData();
+         Map<String, String> newMap = newObj.getData();
+         Set<Map.Entry<String,String>> olds = oldMap.entrySet();
+         Set<Map.Entry<String,String>> news = newMap.entrySet();
+         Set<Map.Entry<String,String>> added,removed,changed;
+         added = new HashSet<Map.Entry<String,String>>(news);
+         // Get added keys. From new set remove
+         // - keys already present in old set
+         added.removeIf(arg0 -> {return oldMap.containsKey(arg0.getKey()); });
+
+         removed = new HashSet<Map.Entry<String,String>>(olds);
+         // Get removed keys. From old set remove
+         // - keys still present in new set
+         removed.removeIf((arg0 -> {return newMap.containsKey(arg0.getKey()); }));
+
+         changed = new HashSet<Map.Entry<String,String>>(olds);
+         // Get the changed keys. From the old set remove
+         // - keys which aren't in the new set (deleted)
+         // - keys which are in the new set with same value (unchanged)
+         changed.removeIf(arg0 -> { return !newMap.containsKey(arg0.getKey()) || arg0.getValue().equals(newMap.get(arg0.getKey())); });
+
+         processConfigMapAndSend(changed, (name,rule) -> {log.info("calling DynamicRuleManagement.updateRule({},{}", name, rule.toString());});
+         processConfigMapAndSend(removed, (name,rule) -> {log.info("calling DynamicRuleManagement.removeRule({},{}", name, rule.toString());});
+         processConfigMapAndSend(added, (name,rule) -> {log.info("calling DynamicRuleManagement.addRule({},{}", name, rule.toString());});
```

Comment:
@rigazilla @jabolina We need to figure out how we want to represent the Rule definition throughout db-syncer. We currently use the `SingleRule` interface as the config object, I guess we also need to have a mapping from the `EagerCacheRuleSpec` to `SingleRule`?

https://github.com/gingersnap-project/db-syncer/pull/37

Reply:
I can do the mapping in the informer if we want to isolate the k8s object there. wdyt?

Reply:
other option is use the generated API classes in the whole application if they fits well

Reply:
> other option is use the generated API classes in the whole application if they fit well

I was thinking about that, but I have two related concerns:

1. Currently the `SingleRule` interface dictates the application.properties used to configure the db-syncer. Having these use the `EagerCacheRuleSpec` values might not be as user friendly
2. The db-syncer might be used by Infinispan, so tightly coupling it's configuration with the EagerCacheRuleSpec doesn't make sense in this case.

Reply:
Yeah, my first thought was a mapping `EagerCacheRuleSpec` -> `SingleRule`.